### PR TITLE
MAINT: interpolate.AAA: improve input validation of `max_terms`

### PR DESCRIPTION
**Pull request by j-bowhay**
Ensures `max_terms` is an integer greater than or equal to 1

### CODE REVIEW

File: scipy/interpolate/_aaa.py [hunk 1]
```diff
@@ -198,6 +199,10 @@ def __init__(self, points, values, *, rtol=None, max_terms=100):
         
         if not np.all(np.isfinite(z)):
             raise ValueError("`points` must be finite.")
+        
+        if max_terms < 1 or not np.issubdtype(type(max_terms), np.integer):
+            raise ValueError("`max_terms` must be an integer value greater than or "
```
Comment (ev-br) [inline scipy/interpolate/_aaa.py:204]: Nit : am not a fan of subdtype(np.integer). This is what operator.index is for

Comment (j-bowhay) [reply scipy/interpolate/_aaa.py:204]: Happy to change, suggesting ditching the  `not np.issubdtype(type(max_terms), np.integer)`?

Comment (ev-br) [reply scipy/interpolate/_aaa.py:204]: exactly. 
```
max_terms = operator.index(max_terms)    # raises if not integer
if max_terms < 0:
   raise ValueError(....)
```

**Review by lucascolley — APPROVED** [review]
thanks Jake!

**Pull request merged**